server/neptune/workflows/internal/deploy/revision/queue/deployer.go

```diff
@@ -84,15 +91,54 @@ func (p *Deployer) Deploy(ctx workflow.Context, requestedDeployment terraformWor
 		return nil, err
 	}
 
-	latestDeployment = requestedDeployment.BuildPersistableInfo()
-	if persistErr := p.persistLatestDeployment(ctx, latestDeployment); persistErr != nil {
-		logger.Error(ctx, "unable to persist deployment, proceeding with in-memory value", key.ErrKey, persistErr)
+	// log error and continue deploys if any of the post deploy task fails
+	if err := p.runPostDeployTasks(ctx, requestedDeployment, scope); err != nil {
+		logger.Error(ctx, "error running post deploy tasks", key.ErrKey, err)
 	}
 
 	// Count this as deployment as latest if it's not a PlanRejectionError which means it is a TerraformClientError
 	// We do this as a safety measure to avoid deploying out of order revision after a failed deploy since it could still
 	// mutate the state file
-	return latestDeployment, err
+	return requestedDeployment.BuildPersistableInfo(), err
+}
+
+func (p *Deployer) runPostDeployTasks(ctx workflow.Context, deployment terraform.DeploymentInfo, scope metrics.Scope) error {
+	if err := p.persistLatestDeployment(ctx, deployment.BuildPersistableInfo()); err != nil {
+		return errors.Wrap(err, "persisting deployment")
+	}
+
+	if err := p.startPRRevisionWorkflow(ctx, deployment, scope); err != nil {
+		return errors.Wrap(err, "starting PR Revision workflow")
+	}
+
+	return nil
+}
+
+func (p *Deployer) startPRRevisionWorkflow(ctx workflow.Context, deployment terraform.DeploymentInfo, scope metrics.Scope) error {
+	version := workflow.GetVersion(ctx, version.SetPRRevision, workflow.DefaultVersion, 1)
+	if version == workflow.DefaultVersion {
+		return nil
+	}
+
+	ctx = workflow.WithChildOptions(ctx, workflow.ChildWorkflowOptions{
+		TaskQueue: prrevision.TaskQueue,
+
+		// configuring this ensures the child workflow will continue execution when the parent workflow terminates
+		ParentClosePolicy: enums.PARENT_CLOSE_POLICY_ABANDON,
+	})
+
+	future := workflow.ExecuteChildWorkflow(ctx, p.PRRevisionWorkflow, prrevision.Request{
+		Repo:     deployment.Repo,
+		Root:     deployment.Root,
+		Revision: deployment.Revision,
+	})
+
+	var childWE workflow.Execution
+	if err := future.GetChildWorkflowExecution().Get(ctx, &childWE); err != nil {
+		scope.SubScope("prrevision.start").Counter(metricsKey.ExecutionErrorMetric).Inc(1)
+		return err
```

Comment:
We're going to error out this function here? Should we just log and stat and just return nil?

Reply:
I wanted to structure this in such a way that we get `error running post deploy task: starting PR Revision workflow: <actual error>` or `error running post deploy task: persisting deployment: <actual error>` so that we know it's a post deploy task. 

To your comment, I've returned the error to the caller and incremented the stat there. Lmk what you think of it